### ISSUE TITLE
Allow vm disks to be resized on update

### ISSUE THREAD

**Issue by ddelnano** [terra-farm/terraform-provider-xenorchestra#178]
When a user creates a VM from a template that contains a disk, the disk will be expanded to the size of the VM's specified size. For example, if you have a Ubuntu 20.04 template that has a 10 GB root volume. If you create a VM with a 5 0GB disk, the template's 10 GB disk will be resized to 50 GBs when copied.

Unfortunately this only works when a VM resource is created. We should allow this to work when a VM's disk size is changed (increased). Note: we cannot shrink a disk safely because we shouldn't know if that would corrupt any file systems on it.

**Comment by michaelkoetter** [comment]
This seems to be work now ... more or less?
When I apply with an increased disk size, the VDI is grown (or at least shows the updated disk size in XO). But my Ubuntu 20.04 VM immediately freezes and disk IO seems to be stuck. When I hard-reset it, the Hard disk is no longer bootable. I was able to reproduce this multiple times, so it wasn't just bad luck.

This is probably more an XO/XCP-ng issue (because the API lets us do it), or maybe even a guest issue, but maybe it's the reason why the XO UI won't let me change the disk size while the VM is running 😄 

![Bildschirmfoto 2022-04-10 um 11 41 08](https://user-images.githubusercontent.com/1581172/162612743-83e1b2bc-50b5-4fe7-9d27-6884077ca325.png)

Here's what happened after I changed disk size from 10 -> 20 GiB using terraform:
 
![Bildschirmfoto 2022-04-10 um 11 48 31](https://user-images.githubusercontent.com/1581172/162612855-657ff985-fcfa-4fe6-aba0-35aeae15aed1.png)

Then after a while:

![Bildschirmfoto 2022-04-10 um 11 51 53](https://user-images.githubusercontent.com/1581172/162612861-e10aaebc-dd10-41ff-a2d6-9063d74bd03c.png)

VM is unbootable after this. DIdn't investigate further and for now just decided to add the following so it should never happen unintentionally:

```hcl
...
    lifecycle {
        ignore_changes = [
            # changes to disk are risky, so we ignore them
            disk,
        ]
    }
...
```

**Comment by ddelnano** [comment]
@michaelkoetter thanks for sharing your testing. I agree that it seems like an XCP-ng or guest issue.

I think terraform shouldn't allow a user to do something dangerous like this so at a bare minimum this shouldn't be allowed when it will cause disk corruption. I'm curious if the resize would work if the VM was shutdown first. If that's the case, then terraform should shutdown the VM, perform the resize and restart it similar to what we do for some VM changes.

**Comment by deefdragon** [comment]
yeah, I think this should error out on a change. This Just nuked several disks that I was testing with. Had to reboot to get them to unfreeze, and when I did, they were not even recognized as existing. (VM self shut down after 30 seconds)

I had snapshots, but it was less than Ideal.

**Comment by kakopedreros** [comment]
Testing my self in VM based in CentOS 7 and Rocky Linux, using Cloud-init, when o create de VM form a 10 GB template, i can resize with issues, only when the VDI its no partitioned in LVM. I try change the RAM an dCPU configuration in hot-swap test and works, but the change of the size of the VDI happen as just @michaelkoetter describe, freeze VM, a bunch of XFS errors and when i restart the VM, its not recognizable as a bootable device.

**Comment by gCyrille** [comment]
Fixed in #357
